### ISSUE TITLE
Allow user to set MinInstanceTypesForSpotToSpotConsolidation

### ISSUE THREAD

**Issue by leoryu** [kubernetes-sigs/karpenter#1645]
### Description

**What problem are you trying to solve?**

At present, the `MinInstanceTypesForSpotToSpotConsolidation` is hard coded with 15.  But sometime the users want the consolidation works enven the candidate spot instance types just one.

https://github.com/kubernetes-sigs/karpenter/blob/140d35b0a4bef7e8914d3cec7efd9aa10945ad52/pkg/controllers/disruption/consolidation.go#L47

**How important is this feature to you?**

I'm building an new provider, I think the `karpenter core` should give users more options when they want the `spot-to-spot` feature.

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jonathan-innis** [comment]
> But sometime the users want the consolidation works enven the candidate spot instance types just one

The reason we opted to not make this user-configurable is covered in the [Original Spot to Spot Consolidation design] but the essence of it is that if you make this number too small, you can race to the bottom and get constantly interrupted. Setting this value to a low value effectively reduces selecting spot instances down to selecting the cheapest spot instance -- which we're a bit weary of since it might cause a lot of application workload disruption.

**Comment by leoryu** [comment]
> I'm hesitant to say that we will accept this -- given the implications. What's your use-case for wanting to change this value and what value do you want to set this to?

Since I have some workloads which are `nice to have`, these workloads are not required to wrok all the time. For this case, I don't want consolidation be blocked by the const, in other words, I want the value is just 1.

I think the default value is good for users to know the potential risks, but it is alseo makes sense to give users the `accepting the risks` option.

**Comment by grosser** [comment]
sounds like a very easy feature to add so users can experiment, so please reopen

usecase is for example "I have 12 instance types I can use and want to see if this feature helps"
